src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CreateCaseCallbackTest.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers;
+
+import io.restassured.response.ValidatableResponse;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.config.IntegrationTest;
+
+import java.io.IOException;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.containing;
+import static com.github.tomakehurst.wiremock.client.WireMock.givenThat;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.google.common.io.Resources.getResource;
+import static com.google.common.io.Resources.toByteArray;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.text.MatchesPattern.matchesPattern;
+import static org.springframework.http.HttpStatus.OK;
+
+@IntegrationTest
+class CreateCaseCallbackTest {
+
+    private static final String CASE_ID_REGEX =
+        // will be replaced with real one after ccd api
+        "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
+
+    private static final String IDAM_TOKEN = "idam-token";
+    private static final String USER_ID = "user-id";
+
+    @LocalServerPort
+    int serverPort;
+
+    @Test
+    void should_create_case_if_classification_new_application_with_documents_and_ocr_data() {
+        setUpTransformation(getTransformationResponseBody("ok-no-warnings.json"));
+
+        postWithBody(getRequestBody("valid-new-application-with-ocr.json"))
+            .statusCode(OK.value())
+            .body("errors", empty())
+            .body("warnings", empty())
+            .body("data.caseReference", matchesPattern(CASE_ID_REGEX));
```

Comment:
I would just assert for not null and not have that regex in place as it will be removed soon.

Reply:
It won't be removed, next PR is just modifying the regex - asserting correct case id